tutor/src/screens/assignment-builder/ux.js

```diff
@@ -78,10 +78,14 @@ class AssignmentBuilderUX {
   }
 
   @computed get isInitializing() {
-
     return !this.isReady || !this.plan || this.plan.api.isPendingInitialFetch;
   }
 
+  @computed get canSelectAllSections() {
+    return Boolean(
+      this.plan.isEditable && this.plan.tasking_plans.length == this.course.periods.length
```

Comment:
🤔 when would the second part be false?

Reply:
if they unselect a section in the builder it removes that tasking plan.  in that case the lengths won't match and this will prevent them from re-selecting "all sections".  We could also check that the tasking_plans period_id matches periods, but this is close enough and much easier to compute